### ISSUE TITLE
Stack smashing errors for creating parallel file on ubuntu with openmpi HDF5

### ISSUE THREAD

**Issue by aniabrown** [AlexJamesWright/METHOD#31]
When running on Ubuntu with OpenMPI 2.1.1, gcc 7.5.0 and HDF5 1.10.0-patch1, creating a new HDF5 file in parallel leads to stack smashing errors. This does not occur when running on Iridis using mpich 3.2.1, gcc 4.8.5 and HDF5 1.10.2. 

As the version on Iridis appears to give the right answer and the fact that the error is occuring withing the H5Fcreate library call, I suspect that this is due to a problem with openmpi, but haven't been able to confirm yet.

**Comment by aniabrown** [comment]
Can now confirm this does depend on MPI version. The code runs with no errors using:

* openmpi 4.0.3, gcc 9.3.0, hdf5 openmpi 1.10.4+repack-11ubuntu1, ubuntu 20.0.4
* mpich 3.3a2, gcc 7.5.0, hdf5 mpich 1.10.0-patch1+docs-4, ubuntu 18.04
* mpich 3.2.1, gcc 4.8.5, hdf5 mpich 1.10.2, RHEL 7.4 (Iridis)

The code throws smashing errors when run on ubuntu 18.04 with the default openmpi 2, however mpich can be installed easily from the package manager on that version of ubuntu. There are therefore enough reasonable options available that I'd suggest listing either mpich 3 or openmpi 4 as a requirement for METHOD rather than trying to track down why openmpi 2 doesn't work.

**Comment by IanHawke** [comment]
Agreed. I will think on this until the next meeting to work out where this is best documented/imposed.
